### PR TITLE
Ansible support on the Magician

### PR DESCRIPTION
**Pull request by rambleraptor**
Ansible support on the Magician.

Change-Id: Ifaf460b447f520508589f9048695ae054b11e221

-----------------------------------------------------------------
# [all]
Ansible support on the Magician.

### CODE REVIEW

File: .ci/ci.yml.tmpl [hunk 1]
```diff
@@ -284,12 +323,26 @@ jobs:
...
               # not push the update even though the commit hashes are different.
               only_if_diff: true
               force: true
         {% endif %}
+        {% if ansible_enabled %}
+          - put: ansible-intermediate
+            params:
+              repository: magic-modules-with-comment/build/ansible
+              branch_file: magic-modules-with-comment/original_pr_branch_name
+              # Every time a change runs through this pipeline, it will generate a commit with
```
Comment (ndmckinley) [inline .ci/ci.yml.tmpl]: Can skip the comment the second time - easy for them to get out of sync.  "See comment on terraform push" will do.

File: .ci/magic-modules/generate-ansible.sh [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+set -e
+
+pushd magic-modules-branched
+LAST_COMMIT_AUTHOR="$(git log --pretty="%an <%ae>" -n1 HEAD)"
+bundle install
+bundle exec compiler -p products/compute -e ansible -o "build/ansible/"
+
+# This command can crash - if that happens, the script should not fail.
+set +e
+ANSIBLE_COMMIT_MSG="$(python .ci/magic-modules/extract_from_pr_description.py --tag ansible < .git/body)"
```
Comment (ndmckinley) [inline .ci/magic-modules/generate-ansible.sh]: Remember to add the `ansible` tag to the .github/PULL_REQUEST_TEMPLATE.md.

File: .ci/ci.yml.tmpl [hunk 3]
```diff
@@ -1,12 +1,13 @@
...
 {% set terraform_enabled = true %}
+{% set ansible_enabled = true %}
 {% macro names_as_list(repo, names) -%}
 {% for name in names %}
 build/{{repo}}/{{name}} 
 {%- endfor %}
 {% endmacro -%}
 {% set puppet_submodules = names_as_list('puppet', puppet_modules).split() %}
-{% set all_submodules = puppet_submodules + ['build/terraform'] if terraform_enabled else [] %}
+{% set all_submodules = puppet_submodules + (['build/terraform'] if terraform_enabled else []) + (['build/ansible'] if ansible_enabled else []) %}
```
Comment (nelsonjr) [inline .ci/ci.yml.tmpl]: Could this fit in less than 500 chars? ;-)

File: .ci/magic-modules/generate-ansible.sh (hunk 2 above)
Comment (nelsonjr) [reply .ci/magic-modules/generate-ansible.sh]: A document for how to add a new provider is desperately needed. Right now is beyond witchcraft how to add a new product or provider. Any chance we can do that?

Comment (ndmckinley) [reply .ci/magic-modules/generate-ansible.sh]: I have an issue to write that up, but it's not near the top of my list right now.  Do you think it should be?

**Pull request merged**